packages/palette/src/elements/FilterSelect/Components/FilterSelectContext.tsx

```diff
@@ -177,6 +193,12 @@ export const FilterSelectContextProvider: React.FC<
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [props.selectedItems?.length])
 
+  useUpdateEffect(() => {
+    if (props.items) {
+      contextValue.setItems(props.items)
+    }
+  }, [props.items, contextValue])
+
```

Comment:
I wonder if most of this stuff could just be taken out of state and computed in `useMemo`s